### PR TITLE
Better overwrite protection

### PR DESCRIPTION
**Pull request by snowskeleton**
Save all downloads to a log file, and check that log file before downloading files. I made this change because I wanted the freedom to delete the aax/aaxc files after conversion, but still wanted to use ```audible download all``` to download a copy only of the books I haven't already downloaded.

### CODE REVIEW

**Comment by mkb79** [comment]
The idea is good! What you are thinking about adding an option to the download command like `--history-file` or specify a history file in the config file?

The reason for this is that `audible-cli` makes use of profiles. I have multiple profiles (manage accounts from family members) which I‘m using. Some books are on multiple accounts for some reasons. This can result in a conflict if I download a book with one profile and try it with another profile again.

**Review by mkb79 — COMMENTED** [review]
Shorten the output is a very good Idea. I will implement this in any cases!

**Comment by snowskeleton** [comment]
I've been thinking about this for a bit, and I think there's a couple ways forward.

Option one is to simply extend the current config file stored in ```~/.audible``` to include a history file per profile, and run checks against the history file specified therein. This is probably the simplest.

Option two is a bit more drastic, and that's to start using something like an SQLite database to keep track of both config and download history. I'm not super fond of this option as it would require much more of a rewrite and adds an additional dependency.

What are you thoughts on this?

**Comment by mkb79** [comment]
Option two has many advantages. There can be a sql db of the entire library, too. Let me think some hours about the best solution.

**Comment by snowskeleton** [comment]
I've been thinking about this a bit more. I like the idea of having a local library database, since it could make a lot of checks easier than making an API call. My biggest hesitation is that it complicates a lot of other, already existing, features. Right now, all of the config files are easily editable by hand. If we use a database, then we lose that functionality. That can be somewhat remedied if audible-cli has commands to help manage the database (```audible database update```, or something).

In short, I think a database has a lot of benefits, but also a lot of extra complication. It's definitely possible, but it would mean lots of refactoring. 

I'm curious what answers you've come up with after having time to think about it. It's definitely not an easy question.

**Comment by mkb79** [comment]
I‘m thinking the same about this. But I would leave the config file untouched and only use the db to store the API responses, the download progress, etc. I decided on a family short break this weekend at short notice! Monday I'll be back again and have more time.

**Comment by snowskeleton** [comment]
Enjoy your weekend! I'll keep playing with this, see if I get something I like.

**Comment by mkb79** [comment]
Thanks. I‘ll have some knowledge with the SQLAlchemy ORM. I think it’s a good solution.

**Comment by snowskeleton** [comment]
See PR #87 for SQLAlchemy integration.
